onnx/optimizer/passes/eliminate_unused_initializer.h

```diff
@@ -0,0 +1,73 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/optimizer/passes/optimize_pass.h"
+
+namespace ONNX_NAMESPACE { namespace optimization {
+
+struct EliminateUnusedInitializer final : public OptimizePass {
+  explicit EliminateUnusedInitializer()
+    : OptimizePass("eliminate_unused_initializer", API_TYPE::IR) {
+  }
+
+  void eliminate_unused_initializer(Graph& graph) {
+    std::unordered_set<std::string> used_initializer_names;
+    // get outputs unique names
+    std::vector<std::string> outputs_unique_names;
+    std::transform(
+        graph.outputs().begin(),
+        graph.outputs().end(),
+        std::back_inserter(outputs_unique_names),
+        [](Value* output) { return output->uniqueName(); });
+    for (auto it = graph.begin(); it != graph.end(); ++it) {
+      auto* n = *it;
+      DescendOnGraphAttributes(n, [this](Graph& g){eliminate_unused_initializer(g);});
```

Comment:
Here, we cannot simply call DescendOnGraphAttributes, because we also need record all the inputs of the nodes in the subgraph.